### PR TITLE
Update metrics for managed keys

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- some of the metrics should be updated when one of the managed keys is leader
  
## Proposed changes
- update metrics

## Testing procedure
- will be tested with feat branch

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/block/metrics.go [hunk 1]
```diff
@@ -161,42 +117,43 @@ func saveMetricsForCommitMetachainBlock(
...
+			appStatusHandler.Increment(common.MetricCountConsensusAcceptedBlocks)
+			continue
 		}
-	}
 
-	if !isInConsensus {
-		return
+		if managedPeersHolder.IsKeyManagedByCurrentNode([]byte(publicKey)) {
+			appStatusHandler.Increment(common.MetricCountConsensusAcceptedBlocks)
+		}
```
Comment (BeniaminDrasovean) [inline process/block/metrics.go:155]: Can be rewritten as:

```
		isMyKey := bytes.Equal([]byte(publicKey), ownPublicKey)
		isManagedByCurrentNode := managedPeersHolder.IsKeyManagedByCurrentNode([]byte(publicKey))
		if isMyKey || isManagedByCurrentNode {
			appStatusHandler.Increment(common.MetricCountConsensusAcceptedBlocks)
		}

```

File: consensus/spos/bls/subroundStartRound.go [hunk 2]
```diff
@@ -159,10 +159,12 @@ func (sr *subroundStartRound) initCurrentRound() bool {
 		msg = " (my turn in multi-key)"
 	}
 	if leader == sr.SelfPubKey() {
+		msg = " (my turn)"
+	}
+	if len(msg) != 0 {
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundStartRound.go:164]: :+1:

File: consensus/spos/bls/subroundStartRound.go [hunk 3]
```diff
@@ -181,10 +183,10 @@ func (sr *subroundStartRound) initCurrentRound() bool {
 		}
 		sr.AppStatusHandler().SetStringValue(common.MetricConsensusState, "not in consensus group")
 	} else {
-		if leader != sr.SelfPubKey() {
+		if leader != sr.SelfPubKey() && !sr.IsKeyManagedByCurrentNode([]byte(leader)) {
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundStartRound.go:186]: good find :+1:

File: process/block/metrics.go [hunk 4]
```diff
@@ -161,42 +117,43 @@ func saveMetricsForCommitMetachainBlock(
...
 
-func countMetaAcceptedSignedBlocks(
-	publicKeys []string,
-	ownPublicKey []byte,
-	appStatusHandler core.AppStatusHandler,
-) {
-	isInConsensus := false
+	for index, publicKey := range pubKeys {
+		if index == 0 {
+			continue
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply process/block/metrics.go:145]: that's the leader index. I guess we can have a const like `leaderIndex := 0` and use that in the if branch

File: process/block/metrics.go [hunk 5]
```diff
@@ -161,42 +117,43 @@ func saveMetricsForCommitMetachainBlock(
...
+	nodesCoordinator nodesCoordinator.NodesCoordinator,
+	appStatusHandler core.AppStatusHandler,
+	managedPeersHolder common.ManagedPeersHolder,
+) {
 	pubKeys, err := nodesCoordinator.GetConsensusValidatorsPublicKeys(
-		header.PrevRandSeed,
-		header.Round,
+		header.GetPrevRandSeed(),
+		header.GetRound(),
 		core.MetachainShardId,
```
Comment (iulianpascalau) [inline process/block/metrics.go:133]: is this ok? Shouldn't have been header.ShardID() ?

File: process/block/metrics.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply process/block/metrics.go:145]: added constant

File: process/block/metrics.go (hunk 1 above)
Comment (sstanculeanu) [reply process/block/metrics.go:155]: good suggestion, applied

File: process/block/metrics.go (hunk 5 above)
Comment (sstanculeanu) [reply process/block/metrics.go:133]: right, messed it up while merging the methods

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**